rdflib/plugins/stores/auditable.py

```diff
@@ -56,9 +56,10 @@ def add(self, triple, context, quoted=False):
             context = context.__class__(self.store, context.identifier) if context is not None else None
             ctxId = context.identifier if context is not None else None
             self.reverseOps.append((s, p, o, ctxId, 'remove'))
-            if (s, p, o, ctxId, 'add') in self.reverseOps:
-                self.reverseOps.remove(
-                    (s, p, o, context, 'add'))
+            try:
+                self.reverseOps.remove((s, p, o, ctxId, 'add'))
+            except ValueError:
+                pass
```

Comment:
In add() there was a bug: it checks if (s, p, o, **ctxId**, 'remove') is in self.reverseOps and if it is found it removes (s, p, o, **context**, 'add')

Then, it is generally a bad idea to check if an element is in a list before removing it. Just try to remove and deal with the ValueError exception.

```python
if x in l:
    l.remove(x)
else:
    l.add(y)
```

becomes:

```python
try:
    l.remove(x)
except ValueError:
    l.add(y)
```
It saves an uneeded call to `__contains__`